server/text_generation_server/utils/layers.py

```diff
@@ -349,14 +350,20 @@ def get_linear(weight, bias, quantize):
             raise NotImplementedError(
                 f"The passed weight is not `awq` compatible, loader needs to be updated."
             )
-        linear = WQLinear(
-            w_bit=bits,
-            group_size=groupsize,
-            qweight=qweight,
-            qzeros=qzeros,
-            scales=scales,
-            bias=bias is not None,
-        )
+        if HAS_AWQ:
```

Comment:
I'm really against that kind of hidden control flow.

Let's make it trivial for users to convert from AWQ to GPTQ externally, and then actually use GTPQ.
Making unasked for, on-the-fly conversions is really not ok.

Reply:
I see, does using exllama as a backend for awq exclusively on rocm make more sense ?

Reply:
I'm not happy of introducing a rocm vs nvidia difference either.

Don' t you think we could have instead a good error message when trying to load AWQ on rocm, and the error message includes a trivial way to use that model anyway.

```
text-generation-launcher --model-id XXX-awq --quantize awq
# Error ! AWQ on Rocm is not supported directly, you can use the GPTQ quantization to use them
text-generation-launcher --model-id XXX-awq --quantize gptq
```

For instance.Wdyt ? It seems quite obvious to users.It might be useful on nvidia targets too, an we can keep the almost transparent feeling.

Reply:
We *do* need to explicit log about the conversion happening though.

Reply:
We can use this flag https://huggingface.co/TheBloke/Mistral-7B-Instruct-v0.2-AWQ/blob/main/config.json#L24 to do the conversion on the fly.